landside/catkin_ws/src/gui/src/gui/launch_dialog.py

```diff
@@ -119,52 +129,153 @@ def traverse_tree(root):
 
             for row in range(len(self.arg_form_rows)):
                 arg = self.arg_form_rows[row]
+                arg['allow_empty'] = True
+
                 if arg.get('default') is None:
                     default_value = ''
                 else:
                     default_value = arg['default']
 
                 label = QtWidgets.QLabel(arg['name'])
+
                 input = QtWidgets.QLineEdit()
                 input.setText(default_value)
 
+                toolTip = ""
+
+                if arg.get("doc") is not None:
```

Comment:
This is a pretty long if-else chain, we should consider extracting some of these conditionals to functions for more clarity.

Reply:
Done